### ISSUE TITLE
vscode macOS c++ debug external console does nothing

### ISSUE THREAD

**Issue by hellblazer33** [microsoft/vscode-cpptools#5343]
Type: Debugger

**Describe the bug**

I am on vscode on mac trying to debug c++ In program where we have to input but it doesnt work as vscode launches external console(terminal) but nothing happens in the terminal.

I am using "externalConsole: true" the terminal launches but debugger doesnt work after launching terminal and nothing happens.

if "externalConsole: false" and program without input then it works perfectly.
I think its problem with latest vscode update for macos.I have windows too and there external console works perfectly while debugging.

- OS and Version: MacOS Catalina 1.44.2
- VS Code Version: 1.44.2
- C/C++ Extension Version: 0.27.0

here is my launch.json config.

```
   {
  "version": "0.2.0",
  "configurations": [
    
    {
      "name": "clang++ - Build and debug active file",
      "type": "cppdbg",
      "request": "launch",
      "program": "${fileDirname}/${fileBasenameNoExtension}",
      "args": [],
      "stopAtEntry": false,
      "cwd": "${workspaceFolder}",
      "showDisplayString": false,
      "environment": [],
      "externalConsole": true,
      "internalConsoleOptions": "openOnSessionStart",
      "MIMode": "lldb",
      "logging": {
        "moduleLoad": false,
        "programOutput": true,
        "trace": false
      },
      "preLaunchTask": "clang++ build active file",
      "osx": {
        "MIMode": "lldb" 
      }
    }
  ```

Here is the code i am trying to debug.
`

```
#include<iostream>
using namespace std;

int main(){
    int a,b,s;
     a=5;
     cout<<"Enter Number";
     cin>>b;
    if((a<200)&&(b<200)){
        s = a + b;
        cout<<s<<endl;
    }
    return 0;
}
```
  

**log**
`
--> R (setFunctionBreakpoints-5): {"type":"response","request_seq":5,"success":true,"command":"setFunctionBreakpoints","body":{"breakpoints":[]},"seq":167}
<--   C (configurationDone-6): {"command":"configurationDone","type":"request","seq":6}
--> E (output): {"type":"event","event":"output","body":{"category":"console","output":"1: (3090) Send Event AD7LoadCompleteEvent\n"},"seq":171}
1: (3090) Send Event AD7LoadCompleteEvent
--> R (configurationDone-6): {"type":"response","request_seq":6,"success":true,"command":"configurationDone","body":{},"seq":170}
--> E (output): {"type":"event","event":"output","body":{"category":"stdout","output":"Warning: Debuggee TargetArchitecture not detected, assuming x86_64.\n"},"seq":174}
Warning: Debuggee TargetArchitecture not detected, assuming x86_64.
--> E (output): {"type":"event","event":"output","body":{"category":"console","output":"1: (3100) <-1012-exec-run\n"},"seq":176}
1: (3100) <-1012-exec-run
<--   C (threads-7): {"command":"threads","type":"request","seq":7}
--> R (threads-7): {"type":"response","request_seq":7,"success":true,"command":"threads","body":{"threads":[]},"seq":179}`

**Comment by Diffumist** [comment]
I have met this problem too.

**Comment by omcnoe** [comment]
I am having the same experience, Terminal.app pops up, but no debug session starts...

**Comment by Diffumist** [comment]
I had the same problem with Manjaro KDE before, but when I changed terminal.external.linuxExec to kconsole (KDE's default terminal) the problem was solved, but it didn't work in Mac OSX, I think it was a problem with VScode's call to terminal.

![image](https://user-images.githubusercontent.com/32810399/81490094-4e132b80-92b0-11ea-9b01-5be8624e5878.png)

**Comment by hellblazer33** [comment]
> I had the same problem with Manjaro KDE before, but when I changed terminal.external.linuxExec to kconsole (KDE's default terminal) the problem was solved, but it didn't work in Mac OSX, I think it was a problem with VScode's call to terminal.
> 
> ![image](https://user-images.githubusercontent.com/32810399/81490094-4e132b80-92b0-11ea-9b01-5be8624e5878.png)

I posted to vscode github too but they said there is no problem with it and its a problem with cpp extension

**Comment by omcnoe** [comment]
This is the the same issue as https://github.com/microsoft/vscode-cpptools/issues/5079, the workaround from that issue https://github.com/microsoft/vscode-cpptools/issues/5079#issuecomment-626090192 worked for me.

**Comment by WenshengGe** [comment]
Same problem on macOS Big Sur.
The only thing working is to launch viscose from a terminal (with code). externalConsole work as expected.

**Comment by pythonreactor** [comment]
> Same problem on macOS Big Sur.
> The only thing working is to launch vscode from a terminal (with code). externalConsole work as expected.

Can you explain what you mean? Maybe throw some steps up? Thanks @WenshengGe

**Comment by LewisDamy** [comment]
> > Same problem on macOS Big Sur.
> > The only thing working is to launch vscode from a terminal (with code). externalConsole work as expected.
> 
> Can you explain what you mean? Maybe throw some steps up? Thanks @WenshengGe

Hey I am having the same problem.
<img width="1370" alt="Screen Shot 2021-01-26 at 12 14 31 AM" src="https://user-images.githubusercontent.com/62454376/105795372-78566780-5f6b-11eb-8ab0-83c267e501fd.png">

When I run the debugger for an C file that contains an input for the user (it does not run, until the user insert it), in VSCode, I've set to open an external terminal and it should be waiting for the user input, but instead, it just launches as usual

**Comment by wrist** [comment]
I also came accross the problem.
At first, I installed `code` command from VSCode command pallet invoked by `Shift+Cmd+p` with the below command.
![image](https://user-images.githubusercontent.com/1150121/108674918-60054880-7529-11eb-802d-c65492f644e7.png)
Then, I executed `code` command from `Terminal.app` to launch new Visual Studio Code instance. 
In such way, `"externalConsole": true` setting worked as expected too.
